Packs/RadwareCloudDDoSProtectionServices/Integrations/RadwareCloudDDoSProtectionServices/RadwareCloudDDoSProtectionServices.py

```diff
@@ -102,86 +104,141 @@ def add_fields(events: list[dict], evnet_type: str | None):
+    args:
+        documents (List[Dict[str, Union[str, Dict]]]): The list of documents.
+        data_type (str): The type of data.
+    Returns: (Tuple[Union[str, None], List[str]]): A tuple containing the latest timestamp and a list of corresponding ids.
+    """
+    if not documents:
+        return None, []
+
+    def get_timestamp(data):
+        if data_type == 'events':
+            return data['endTimestamp']
+        else:
+            return data.get('context', {}).get('_timestamp')
 
-    return documents, last_run
+    latest_timestamp = get_timestamp(documents[0])
+    ids = [documents[0].get('_id')]
 
+    for doc in documents[1:]:
+        if get_timestamp(doc) == latest_timestamp:
+            ids.append(doc.get('_id'))
+        else:
+            break
 
-def fetch_alerts(client, last_run):
+    return latest_timestamp, ids
+
+
+def fetch_data(client, last_run, data_type):
+    """
+    Fetch data of a specified type (either 'events' or 'alerts') from the client.
+
+    Args:
+        client (Client): The client instance to fetch the data from.
+        last_run (dict): the last run data.
+        data_type (str): The type of data to fetch ('events' or 'alerts').
+
+    Returns:
+        tuple: Contains the fetched documents and updated last run dictionary.
+    """
     end_time = int(time.time() * 1000)
-    start_time = last_run.get('last_fetch_alerts', end_time-2)+1
-    continue_fetch_alerts = last_run.get('continue_fetch_alerts', None)
+    last_fetch_key = f'last_fetch_{data_type}'
+    continue_fetch_key = f'continue_fetch_{data_type}'
+
+    latest_fetch = last_run.get(last_fetch_key, {})
+    start_time = latest_fetch.get('latest_timestamp', end_time - 1)
+    continue_fetch_data = last_run.get(continue_fetch_key, None)
     skip = 0
 
-    if continue_fetch_alerts:
-        end_time = continue_fetch_alerts.get('end_time')
-        start_time = continue_fetch_alerts.get('start_time')
-        skip = continue_fetch_alerts.get('fetched_alerts')
+    if continue_fetch_data:
+        end_time = continue_fetch_data.get('end_time')
+        start_time = continue_fetch_data.get('start_time')
+        skip = continue_fetch_data.get('fetched_' + data_type)
+
+    demisto.debug(f'RadwareCloudDDoS: {data_type=} {start_time=}, {end_time=}, {skip=}')
+
+    if data_type == 'events':
+        response = client.get_events(start_time, end_time, skip, PAGE_SIZE)
+    elif data_type == 'alerts':
+        response = client.get_alerts(start_time, end_time, skip, PAGE_SIZE)
+    else:
+        raise ValueError("Invalid data_type. Expected 'events' or 'alerts'.")
 
-    demisto.debug(f'RadwareCloudDDoS: {start_time=}, {end_time=},  {skip=}')
-    response = client.get_alerts(start_time, end_time, skip, 700)
     documents = response.get("documents")
-    demisto.debug(f'RadwareCloudDDoS: {len(documents)=}')
-    new_continue_fetch_alerts = {}
+    demisto.debug(f'RadwareCloudDDoS: {data_type=} {len(documents)=}')
```

Comment:
replace RadwareCloudDDoS with relevant string or remove
```suggestion
    demisto.debug(f'RadwareCloudDDoS: {data_type=} {len(documents)=}')
```